test/use-swr-integration.test.tsx

```diff
@@ -470,4 +470,103 @@ describe('useSWR', () => {
     await act(() => sleep(150))
     expect(count).toBe(2)
   })
+
+  // Test for https://swr.vercel.app/docs/advanced/performance#dependency-collection
+  it('should render four times in the worst case', async () => {
```

Comment:
https://github.com/reactwg/react-18/discussions/17
we could use `<Profiler />` to count the number of times a component rendered. Then the test will still work in strict mode. 

https://github.com/vercel/swr/blob/8bfa5da876b4948bdaec5359479c0aa062ee74fb/test/use-swr-integration.test.tsx#L458-L465

Reply:
@promer94 
Thank you! I didn't know the component 👍 

In this test, I'd like to assert the return values too, and the test requires reading return values within rendering so that SWR can mark the values as used because it's a test for dependency collection.